cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -130,6 +133,8 @@
   private final ProvisioningService provisioningService;
   private final ProgramStateWriter programStateWriter;
   private final int maxConcurrentRuns;
+  private static final String RUNTIME_ARG_KEYTAB = "pipeline.keytab.path";
```

Comment:
1. Please move system runtime arguments constant definition to `SystemArguments` class.
1. System arguments should be prefixed with `system.`.
1. I believe the argument is not pipeline specific. Please name it as `system.kerberos.principal.name` and `system.kerberos.keytab.path`.
1. Also, what's the requirement of the `path`? Is it a path on HDFS? Does it support a fully qualified URI as well?

Reply:
Arguments are pipeline specific. Path is local FS path on CDAP master node of keytab file.

Reply:
I have moved the constant definition to io.cdap.cdap.internal.app.runtime.ProgramOptionConstants file

Reply:
Why it is pipeline specific? It cannot apply to all program types??

Reply:
The requirement is for providing ability to configure impersonation at runtime for pipelines. I haven't tested other program types. If you feel, the changes are applicable to other types then I can rename the configuration appropriately.

Reply:
The changes involved should be applicable to all program types (pipeline is actually not a program type).

Reply:
Changed the property names to "runtime.principal.name" and "runtime.keytab.path". Please check and merge.